doc/Streaming.xml

```diff
@@ -590,7 +605,245 @@
+            <para>
+              The device may list which SRTP cryptographic algorithms it supports through the Media2 wsdl.
+              Each encoder can list the algorithms it supports through the SecureStreamingProtocolAlgorithms element present in:
+              <itemizedlist>
+                <listitem>
+                  <para>AudioEncoder2ConfigurationOptions</para>
+                </listitem>
+                <listitem>
+                  <para>VideoEncoder2ConfigurationOptions</para>
+                </listitem>
+                <listitem>
+                  <para>MetadataConfigurationOptions</para>
+                </listitem>
+              </itemizedlist>
+
+              If the SecureStreamingProtocolAlgorithms parameter is present, the device shall support at least one of the following algorithms:
+              <itemizedlist>
+                <listitem>
+                  <para>NONE</para>
+                  <para>NONE is a special case, where the media is listed as RTP/AVP instead of RTP/SAVP and where MIKEY is unused altogether.</para>
+                  <para>
+                    This it to allow the use of RTSPS without using an SRTP encrypted media stream.
+                    It can also be used for RTSPS interleaved where media is sent over the same TLS protected channel as the RTSP control messages.
+                  </para>
+                </listitem>
+                <listitem>
+                  <para>AES_CM_128_HMAC_SHA1_80</para>
+                </listitem>
+                <listitem>
+                  <para>AEAD_AES_128_GCM</para>
+                </listitem>
+                <listitem>
+                  <para>AEAD_AES_256_GCM</para>
+                </listitem>
+              </itemizedlist>
+              If the device supports SRTP streaming through the SecureRTSPStreaming parameter, but does not list any supported SRTP cryptographic algorithms through the SecureStreamingProtocolAlgorithms parameter, the device shall support the AES_CM_128_SHA1_80 algorithm as defined in RFC 3711.
+            </para>
+
+            <para>
+              The client may choose to configure a device to use a specific algorithm. To do so, it shall set the SecureStreamingProtocolAlgorithm attribute through the following methods:
+              <itemizedlist>
+                <listitem>
+                  <para>SetAudioEncoder2Configuration</para>
+                </listitem>
+                <listitem>
+                  <para>SetVideoEncoder2Configuration</para>
+                </listitem>
+                <listitem>
+                  <para>SetMetadataConfiguration</para>
+                </listitem>
+              </itemizedlist>
+            </para>
+
+            <para>
+              If a device supports encryption algorithms other than AES_CM_128_SHA1_80 as defined in RFC 3711, it shall also support the ConfigurationOptions and the methods listed above. 
```

Comment:
This text can be deleted as the requirement text is optimized in above suggestion.
```suggestion

```